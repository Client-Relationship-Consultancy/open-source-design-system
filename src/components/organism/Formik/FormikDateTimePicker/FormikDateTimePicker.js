import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import Label from "../../../atoms/Label/Label"
import { DateTimePicker } from "../../../molecules/DateTimePicker/DateTimePicker"

export const ErrorMessage = styled.div`
  font-weight: bold;
  color: ${props => props.theme.error.main.hex};
  margin-top: 0.5rem;
`
ErrorMessage.displayName = "ErrorMessage"

const FormikDateTimePicker = props => {
  const { id, labelMessage, value, onChange, errorMessage, touched, ...others } = props
  return (
    <>
      <Label htmlFor={id}>{labelMessage}</Label>
      <DateTimePicker id={id} selectedDate={value} onChange={onChange} {...others} />
      {errorMessage && touched && <ErrorMessage>{errorMessage}</ErrorMessage>}
    </>
  )
}

FormikDateTimePicker.propTypes = {
  errorMessage: PropTypes.string,
  id: PropTypes.string,
  labelMessage: PropTypes.object,
  value: PropTypes.instanceOf(Date),
  onChange: PropTypes.func,
  touched: PropTypes.bool,
}

export default FormikDateTimePicker
