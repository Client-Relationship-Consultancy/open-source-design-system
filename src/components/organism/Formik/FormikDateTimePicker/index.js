import FormikDateTimePikcer from "./FormikDateTimePicker"

export default FormikDateTimePikcer
