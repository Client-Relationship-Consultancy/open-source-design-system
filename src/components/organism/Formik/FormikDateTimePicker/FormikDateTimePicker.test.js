import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import "jest-styled-components"

import FormikDateTimePicker, { ErrorMessage } from "./FormikDateTimePicker"

describe("error message", () => {
  it("should match snapshot", () => {
    const theme = {
      error: {
        main: {
          hex: "#D00000",
        },
      },
    }
    const component = shallow(<ErrorMessage theme={theme} />)
    expect(component).toMatchSnapshot()
  })
})

describe("FormikDateTimePicker", () => {
  const mockFn = jest.fn()
  const message = <div>testing</div>
  const date = new Date("August 25, 1987 11:13:00 GMT+0000 (UTC)")

  const component = shallow(
    <FormikDateTimePicker
      errorMessage=""
      touched={false}
      id="date"
      labelMessage={message}
      value={date}
      onChange={mockFn}
    />,
  )
  it("should match snapshot", () => {
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
  it("should call mock function when onChange", () => {
    const now = new Date("August 25, 1987 11:13:00 GMT+0000 (UTC)")
    component.find("DateTimePicker").simulate("change", now)
    expect(mockFn).toHaveBeenCalledWith(now)
  })

  it("should have error component if error exists", () => {
    expect(component.exists("ErrorMessage")).toBeFalsy()
    component.setProps({ errorMessage: "there is an error", touched: true })
    expect(component.exists("ErrorMessage")).toBeTruthy()
  })
})
