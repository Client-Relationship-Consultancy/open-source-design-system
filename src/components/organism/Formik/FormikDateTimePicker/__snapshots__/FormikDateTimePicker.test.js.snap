// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikDateTimePicker should match snapshot 1`] = `
<Fragment>
  <Label
    htmlFor="date"
  >
    <div>
      testing
    </div>
  </Label>
  <DateTimePicker
    id="date"
    onChange={[MockFunction]}
    selectedDate={1987-08-25T11:13:00.000Z}
    showTimeSelect={true}
  />
</Fragment>
`;

exports[`error message should match snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "FormikDateTimePicker__ErrorMessage-ep3qoy-0",
        "isStatic": false,
        "rules": Array [
          "font-weight:bold;color:",
          [Function],
          ";margin-top:0.5rem;",
        ],
      },
      "displayName": "ErrorMessage",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "FormikDateTimePicker__ErrorMessage-ep3qoy-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "error": Object {
        "main": Object {
          "hex": "#D00000",
        },
      },
    }
  }
/>
`;
